sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ModuleIdentityLifecycleSamples.cs

```diff
@@ -120,34 +119,149 @@ public async Task<ModuleIdentity> CreateModuleIdentityAsync(DeviceIdentity devic
+                moduleIdentity.ManagedBy = Environment.UserName;
+
+                Response<ModuleIdentity> response = await IoTHubServiceClient.Modules.CreateOrUpdateIdentityAsync(moduleIdentity, IfMatchPrecondition.UnconditionalIfMatch);
+
+                ModuleIdentity updatedModule = response.Value;
+                
+                SampleLogger.PrintSuccess($"Successfully updated module identity with Id: '{updatedModule.ModuleId}', ManagedBy: '{updatedModule.ManagedBy}', ETag: '{updatedModule.Etag}'");
+                
+                return updatedModule;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to update a module identity due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task UpdateModuleTwinAsync()
+        /// <summary>
+        /// Get module twin.
+        /// </summary>
+        /// <param name="moduleIdentity">The module identity.</param>
+        public async Task<TwinData> GetModuleTwinAsync(ModuleIdentity moduleIdentity)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("GET A MODULE TWIN");
+
+            try
+            {
+                Console.WriteLine($"Getting module twin with Id: '{moduleIdentity.ModuleId}'\n");
+
+                Response<TwinData> response = await IoTHubServiceClient.Modules.GetTwinAsync(moduleIdentity.DeviceId, moduleIdentity.ModuleId);
+
+                SampleLogger.PrintSuccess($"\t- Module Twin Status: '{response.Value.Status}', ETag: '{response.Value.Etag}'");
+
+                return response.Value;
+            }
+            catch (Exception ex)
+            {
+                SampleLogger.FatalError($"Failed to get a module twin due to:\n{ex}");
+                throw;
+            }
         }
 
-        public Task InvokeMethodAsync()
+        /// <summary>
+        /// Update a module twin desired properties
+        /// </summary>
+        /// <param name="moduleTwin">Module twin.</param>
+        public async Task<TwinData> UpdateModuleTwinAsync(TwinData moduleTwin)
         {
-            return Task.CompletedTask;
+            SampleLogger.PrintHeader("UPDATE A MODULE TWIN");
+
+            string userPropName = "user";
+
+            try
+            {
+                Console.WriteLine($"Updating module twin with Id: '{moduleTwin.ModuleId}'. Setting Desired property {userPropName} to: '{Environment.UserName}'\n");
+
+                moduleTwin.Properties.Desired.Add(new KeyValuePair<string, object>(userPropName, Environment.UserName));
+
+                Response<TwinData> response = await IoTHubServiceClient.Modules.UpdateTwinAsync(moduleTwin, IfMatchPrecondition.UnconditionalIfMatch);
+
+                TwinData updatedTwin = response.Value;
+
+                var userPropValue = (string) updatedTwin.Properties.Desired
+                    .Where(p => p.Key == userPropName)
```

Comment:
Do we need a null check here? What happens if `.Where(p => p.Key == userPropName)` returns 0 entries?

Reply:
it should fatally fail. That should never happen.

Reply:
sounds good!